src/3rd_party_update.c

```diff
@@ -260,7 +301,40 @@ enum swupd_code third_party_update_main(int argc, char **argv)
 
 	/* update 3rd-party bundles */
 	ret_code = third_party_run_operation_multirepo(cmdline_option_repo, update_repos, SWUPD_NO, "update", steps_in_update);
+	if (ret_code) {
+		goto exit;
+	}
+
+	/* read the current template copy */
+	template_file = sys_path_join("%s/%s/%s", globals_bkp.path_prefix, SWUPD_3RD_PARTY_DIR, SWUPD_3RD_PARTY_TEMPLATE_FILE);
+	ret = map_file_into_memory(template_file, &template, &template_len);
+	if (ret && ret != ENOENT) {
+		error("The scripts template file %s could not be read\n", template_file);
+		goto exit;
+	}
+
+	if (!template || strcmp((char *)template, SCRIPT_TEMPLATE) != 0) {
+		/* there is no template file, or the template changed,
+		 * all scripts need to be recreated */
+		info("The scripts that export binaries from 3rd-party repositories need to be regenerated\n\n");
+		ret_code = third_party_run_operation_multirepo(NULL, regenerate_all_wrapper_scripts, SWUPD_OK, "regenerate_scripts", steps_in_update);
+
+		/* update the template */
+		ret_code = third_party_bin_directory_exist();
+		if (ret_code) {
+			goto exit;
+		}
+
+		ret = write_into_file(template_file, SCRIPT_TEMPLATE, "w");
```

Comment:
If you write the file here the first update will regenerate wrappers even if not needed
```
$ Swupd 3rd-party add ...
$ swupd 3rd-party bundle-add ...
# there's no template
$ swupd 3rd-party update # First time you run update you always regenerate.
```

Reply:
if there is no template the first update will regenerate the wrappers, I think this behavior is correct, the wrapper don't need to be regenerated at that point but I don't see the harm on it either, it happens only once. What are you suggesting?

Reply:
that's true. it's just in the first update. it's fine then